dask/array/core.py

```diff
@@ -4256,7 +4267,7 @@ def stack(seq, axis=0):
 
     if not seq:
         raise ValueError("Need array(s) to stack")
-    if not all(x.shape == seq[0].shape for x in seq):
+    if not allow_unknown_chunksizes and not all(x.shape == seq[0].shape for x in seq):
```

Comment:
I'm not sure this line is quite right. This will fail to raise

```python
>>> a = da.ones((4, 2))
>>> b = da.ones((5, 2))
>>> da.stack([a, b], allow_unknown_chunksizes)
```

So we're ignoring known chunksizes.

Reply:
Ah, good catch. I'll push a quick fix up now.

Reply:
Actually, this does error on the call to `unify_chunks`. Other dask array methods with the same feature exhibit the same behavior (e.g. `da.concatenate`, `da.vstack`, etc... all have the same logic around `allow_unknown_chunksizes`. As this isn't any worse than the implementation in `da.concatenate`, I'm fine leaving things as is for now.

```python
In [10]: a = da.ones((4, 2))

In [11]: b = da.ones((5, 2))

In [12]: da.stack([a, b], allow_unknown_chunksizes=True)
...
ValueError: Chunks do not add up to shape. Got chunks=((5,), (2,)), shape=(4, 2)

In [13]: da.concatenate([a, b], axis=1, allow_unknown_chunksizes=True)
...
ValueError: Chunks do not add up to shape. Got chunks=((5,), (2,)), shape=(4, 2)
```

Reply:
Sounds good, thanks for checking!